cachi2/core/package_managers/pip.py

```diff
@@ -97,51 +97,57 @@ class DistributionPackageInfo:
     index_url: str
     is_yanked: bool
 
-    pypi_checksums: set[ChecksumInfo] = field(default_factory=set)
-    user_checksums: set[ChecksumInfo] = field(default_factory=set)
+    # PyPi only returns a single checksum for a given download artifact
+    pypi_checksum: set[ChecksumInfo] = field(default_factory=set)
+    # "User" checksums *must* come from a 'requirements*.txt' file or equivalent
+    req_file_checksums: set[ChecksumInfo] = field(default_factory=set)
 
-    checksums_to_verify: set[ChecksumInfo] = field(init=False, default_factory=set)
+    checksums_to_match: set[ChecksumInfo] = field(init=False, default_factory=set)
 
     def __post_init__(self) -> None:
-        self.checksums_to_verify = self._determine_checksums_to_verify()
+        self.checksums_to_match = self._determine_checksums_to_match()
 
-    def _determine_checksums_to_verify(self) -> set[ChecksumInfo]:
-        """Determine the set of checksums to verify for a given distribution package."""
+    def _determine_checksums_to_match(self) -> set[ChecksumInfo]:
+        """Determine the set of checksums to match for a given distribution package."""
         checksums: set[ChecksumInfo] = set()
-        matching: set[ChecksumInfo] = self.pypi_checksums.intersection(self.user_checksums)
-
-        if self.pypi_checksums and self.user_checksums:
-            checksums = matching
-            msg = "using intersection of user specified and PyPI reported checksums"
-        elif self.pypi_checksums:
-            checksums = self.pypi_checksums
-            msg = "using PyPI reported checksums"
-        elif self.user_checksums:
-            checksums = self.user_checksums
-            msg = "using user specified checksums"
+        intersection: set[ChecksumInfo] = self.pypi_checksum.intersection(self.req_file_checksums)
```

Comment:
Nit: the intersection could be calculated in the if block (L116). I don't think having it here improves the readability, but that's just me.